src/oge/validation.py

```diff
@@ -52,64 +52,98 @@ def check_allocated_gf_matches_input_gf(year, gen_fuel_allocated):
     from the total input generation or fuel.
     """
     gf = load_data.load_pudl_table("denorm_generation_fuel_combined_eia923", year)
-    plant_total_gf = gf.groupby("plant_id_eia")[
+    plant_total_gf = gf.groupby(["plant_id_eia", "energy_source_code"])[
         [
             "net_generation_mwh",
             "fuel_consumed_mmbtu",
             "fuel_consumed_for_electricity_mmbtu",
         ]
     ].sum()
-    plant_total_alloc = gen_fuel_allocated.groupby("plant_id_eia")[
+
+    plant_total_alloc = gen_fuel_allocated.groupby(
+        ["plant_id_eia", "energy_source_code"]
+    )[
         [
             "net_generation_mwh",
             "fuel_consumed_mmbtu",
             "fuel_consumed_for_electricity_mmbtu",
         ]
     ].sum()
-    # calculate the percentage difference between the values
-    # replace 0s with small sentinel value to prevent missing values from divide by zero
-    plant_total_diff = (
-        (plant_total_alloc - plant_total_gf) / plant_total_gf.replace(0, 0.00001)
+
+    id2ba = create_plant_ba_table(year).set_index("plant_id_eia")["ba_code"].to_dict()
+
+    symmetric_difference = set(plant_total_gf.index).symmetric_difference(
+        set(plant_total_alloc.index)
+    )
+    if len(symmetric_difference) > 0:
+        logger.warning(
+            f"There are {len(symmetric_difference)} plant/fuel combinations that are "
+            "either in EIA-923 Input or allocated EIA-923 but not in both"
+        )
+        input_minus_allocated = list(
+            set(plant_total_gf.index).difference(set(plant_total_alloc.index))
+        )
+        if len(input_minus_allocated) > 0:
+            logger.warning(
+                f"There are {len(input_minus_allocated)} plant/fuel combinations in "
+                "EIA-923 Input but not in allocated EIA-923 : \n"
+                + plant_total_gf.loc[input_minus_allocated]
+                .assign(ba_code=[id2ba[i] for i, _ in input_minus_allocated])
+                .to_string()
+            )
+        allocated_minus_input = list(
+            set(plant_total_alloc.index).difference(set(plant_total_gf.index))
+        )
+        if len(allocated_minus_input) > 0:
+            logger.warning(
+                f"There are {len(allocated_minus_input)} plant/fuel combinations in "
+                "allocated EIA-923 but not in EIA-923 Input: \n"
+                + plant_total_alloc.loc[allocated_minus_input]
+                .assign(ba_code=[id2ba[i] for i, _ in allocated_minus_input])
+                .to_string()
+            )
+
+    # Align the 2 data frames. Values in missing rows in either data frames are set to 0.
+    plant_total_gf, plant_total_alloc = plant_total_gf.align(
+        plant_total_alloc, fill_value=0
+    )
+    # Calculate the percentage difference between the values
+    # Replace 0s with small sentinel value to prevent missing values from divide by zero
+    plant_total_diff = (plant_total_alloc - plant_total_gf) / plant_total_gf.replace(
```

Comment:
Here, when calculating the percentage diff, we now want to fill na with zero before doing the replace so that we flag where there is a non-zero percentage diff.